tests/xlsx/test_excel_writer.py

```diff
@@ -129,3 +132,87 @@ def test_write_xlsx__correctly_handles_custom_twitch_widths(mocker):
 
         # switch dir back to avoid causing issues with other tests
         os.chdir(cwd)
+
+
+@pytest.mark.parametrize("test_value", [None, False])
+def test_write_xlsx__correctly_handles_include_stim_protocols_param_with_false_values(mocker, test_value):
+    # mock so slow function doesn't actually run
+    mocker.patch.object(
+        magnet_finding,
+        "get_positions",
+        autospec=True,
+        side_effect=lambda x, **kwargs: {"X": np.zeros((x.shape[-1], 24))},
+    )
+
+    pr = PlateRecording(TEST_FILE_WITH_PROTOCOLS)
+
+    # save dir before switching to temp dir
+    cwd = os.getcwd()
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # switch to temp dir so output file is automatically deleted
+        os.chdir(tmpdir)
+        output_file_name = write_xlsx(pr, include_stim_protocols=test_value)
+
+        output_filepath = os.path.join(tmpdir, output_file_name)
+        df = pd.read_excel(output_filepath, None)
+
+        # check that all sheets are present except stimulation-protocols sheet
+        assert len(df.keys()) == 8
+
+        # switch dir back to avoid causing issues with other tests
+        os.chdir(cwd)
+
+
+def test_write_xlsx__correctly_handles_include_stim_protocols_param_with_stim_protocols(mocker):
+    # mock so slow function doesn't actually run
+    mocker.patch.object(
+        magnet_finding,
+        "get_positions",
+        autospec=True,
+        side_effect=lambda x, **kwargs: {"X": np.zeros((x.shape[-1], 24))},
+    )
+
+    pr = PlateRecording(TEST_FILE_WITH_PROTOCOLS)
+
+    # save dir before switching to temp dir
+    cwd = os.getcwd()
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # switch to temp dir so output file is automatically deleted
+        os.chdir(tmpdir)
+        output_file_name = write_xlsx(pr, include_stim_protocols=True)
+
+        output_filepath = os.path.join(tmpdir, output_file_name)
+        df = pd.read_excel(output_filepath, sheet_name="stimulation-protocols", usecols=[1])
+
+        # check that stimulation-protocols sheet has unassigned well lables
+        df.keys()[0] == "D5, D6,"
```

Comment:
This is missing the `assert` keyword. We also probably want to make some more assertions, but I'd say just making sure that each well is paired up with its assigned protocol correctly is fine for now